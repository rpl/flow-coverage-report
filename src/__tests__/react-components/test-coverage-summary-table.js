'use babel';

import React from 'react';
import renderer from 'react-test-renderer';

import {FLOW_COVERAGE_SUMMARY_DATA} from '../fixtures';

import {BASE_DIR} from './common';

const REACT_COMPONENT = `${BASE_DIR}/coverage-summary-table`;

test('<FlowCoverageSummaryTable />', () => {
  const FlowCoverageSummaryTable = require(REACT_COMPONENT).default;

  // Expect positive with lower threshold.
  const positiveSummaryProps = {
    coverageSummaryData: {...FLOW_COVERAGE_SUMMARY_DATA, threshold: 40}
  };
  const positiveSummaryTree = renderer.create(<FlowCoverageSummaryTable {...positiveSummaryProps}/>).toJSON();
  expect(positiveSummaryTree).toMatchSnapshot();

  // Expect positive with higher threshold.
  const negativeSummaryProps = {
    coverageSummaryData: {...FLOW_COVERAGE_SUMMARY_DATA, threshold: 90}
  };
  const negativeSummaryTree = renderer.create(<FlowCoverageSummaryTable {...negativeSummaryProps}/>).toJSON();
  expect(negativeSummaryTree).toMatchSnapshot();
});

test.todo('<FlowCoverageSummaryTable /> with missing props');
