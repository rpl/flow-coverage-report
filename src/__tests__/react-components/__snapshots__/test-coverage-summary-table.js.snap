// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlowCoverageSummaryTable /> 1`] = `
<table
  className="ui small celled table"
>
  <thead>
    <tr>
      <th>
        Percent
      </th>
      <th>
        Total
      </th>
      <th>
        Covered
      </th>
      <th>
        Uncovered
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="positive"
    >
      <td>
        50
         %
      </td>
      <td>
        10
      </td>
      <td>
        5
      </td>
      <td>
        5
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<FlowCoverageSummaryTable /> 2`] = `
<table
  className="ui small celled table"
>
  <thead>
    <tr>
      <th>
        Percent
      </th>
      <th>
        Total
      </th>
      <th>
        Covered
      </th>
      <th>
        Uncovered
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="negative"
    >
      <td>
        50
         %
      </td>
      <td>
        10
      </td>
      <td>
        5
      </td>
      <td>
        5
      </td>
    </tr>
  </tbody>
</table>
`;
